java/pattern/constraint/TypeConstraint.java

```diff
@@ -562,7 +577,24 @@ public boolean isPlays() {
 
         @Override
         public String toString() {
-            return PLAYS.toString() + SPACE + roleType;
+            StringBuilder syntax = new StringBuilder();
+
+            syntax.append(PLAYS).append(SPACE);
+            if (relationType!=null) {
+                if (!relationType.label().isPresent()) syntax.append(relationType);
+                else syntax.append(relationType.label().get().label());
+                syntax.append(COLON);
+            }
+
+            if (!roleType.label().isPresent()) syntax.append(roleType);
+            else syntax.append(roleType.label().get().label());
```

Comment:
This block is also entirely not needed. The old code covers this logic.

Reply:
Done, reduced to the old style while keeping the fix.